compiler.js

```diff
@@ -360,6 +357,67 @@ class SlangCompiler
         }
     }
 
+    getBindingDescriptor(index, programReflection, parameter)
+    {
+        const globalLayout = programReflection.getGlobalParamsTypeLayout();
+
+        const bindingType = globalLayout.getDescriptorSetDescriptorRangeType(0, index);
+
+        // Special case.. TODO: Remove this as soon as the reflection API properly reports write-only textures.
+        if (parameter.getName() == "outputTexture")
+        {
+            return { storageTexture: {access: "write-only", format: "rgba8unorm"} };
+        }
+
+        if (bindingType == this.slangWasmModule.BindingType.Texture)
+        {
+            return { texture: {} };
+        }
+        else if (bindingType == this.slangWasmModule.BindingType.MutableTexture)
+        {
+            return { storageTexture: {access: "read-write", format: "r32float"} };
+        }
+        else if (bindingType == this.slangWasmModule.BindingType.ConstantBuffer)
+        {
+            return { buffer: {type: 'uniform'} };
+        }
+        else if (bindingType == this.slangWasmModule.BindingType.MutableTypedBuffer)
+        {
+            return { buffer: {type: 'storage'} };
+        }
+        else if (bindingType == this.slangWasmModule.BindingType.MutableRawBuffer)
+        {
+            return { buffer: {type: 'storage'} };
+        }
+    }
+
+    getResourceBindings(linkedProgram)
+    {
+        const reflection = linkedProgram.getLayout(0); // assume target-index = 0
```

Comment:
as mentioned in the slang PR, every object that is `new`d from C++ and returned to javascript needs to be explicitly `.delete()`'d in javascript, which is a big pain for small objects like this. We should change the binding to not `new` stuff and just return a raw pointer.

Reply:
Changed this in the Slang PR